mysql-test/suite/versioning/t/truncate_violation.test

```diff
@@ -0,0 +1,27 @@
+create or replace table t1 (id int);
+create or replace table t2 (id int) with system versioning;
+
+-- echo  # force cleaning table shares
+flush tables t1, t2;
+
+truncate table t1;
+--error ER_TRUNCATE_ILLEGAL_VERS
+truncate table t2;
+
+-- echo # fetch table shares
+describe t1;
+describe t2;
+
+truncate table t1;
+--error ER_TRUNCATE_ILLEGAL_VERS
+truncate table t2;
+
+--echo # enter locked tables mode
+lock tables t1 WRITE, t2 WRITE;
+
+truncate t1;
+--error ER_TRUNCATE_ILLEGAL_VERS
+truncate t2;
+
+unlock tables;
+
```

Comment:
Just add these cases to `truncate.test`. No need to create separate test for this.

Reply:
As far as I understand, `versioning/truncate.test` is not directly about `TRUNCATE` command (it is not used there at all).

Reply:
It's better to create less, but larger test files than many small tests, because initialization of each test takes substantial time. This particular case does not deserve separate file.

Reply:
Make comment with `--echo # MDEV-...` to indicate what these cases do.